src/ssl.c

```diff
@@ -8023,15 +8085,91 @@ WOLFSSL_X509_EXTENSION* wolfSSL_X509_set_ext(WOLFSSL_X509* x509, int loc)
 int wolfSSL_X509V3_EXT_print(WOLFSSL_BIO *out, WOLFSSL_X509_EXTENSION *ext,
         unsigned long flag, int indent)
 {
+    ASN1_OBJECT* obj;
+    ASN1_STRING* str;
+    int nid;
+    int sz = CTC_NAME_SIZE*2;
     int rc = WOLFSSL_FAILURE;
-    char tmp[CTC_NAME_SIZE*2];
+    char tmp[sz];
+    WOLFSSL_ENTER("wolfSSL_X509V3_EXT_print");
 
-    if ((out != NULL) && (ext != NULL)) {
+    if ((out == NULL) || (ext == NULL)) {
+        WOLFSSL_MSG("NULL parameter error");
+        return rc;
+    }
 
-        XSNPRINTF(tmp, CTC_NAME_SIZE*2, "%*s%s", indent, "", ext->value.strData);
-        if (wolfSSL_BIO_write(out, tmp, (int)XSTRLEN(tmp)) == (int)XSTRLEN(tmp)) {
-            rc = WOLFSSL_SUCCESS;
-        }
+    obj = wolfSSL_X509_EXTENSION_get_object(ext);
+    if (obj == NULL) {
+        WOLFSSL_MSG("Error getting ASN1_OBJECT from X509_EXTENSION");
+        return rc;
+    }
+
+    str = wolfSSL_X509_EXTENSION_get_data(ext);
+    if (obj == NULL) {
+        WOLFSSL_MSG("Error getting ASN1_STRING from X509_EXTENSION");
+        return rc;
+    }
+
+    /* Print extension based on the type */
+    nid = wolfSSL_OBJ_obj2nid(obj);
+    switch(nid) {
+        case BASIC_CA_OID:
+            {
+                char isCa[] = "TRUE";
+                char notCa[] = "FALSE";
+                XSNPRINTF(tmp, sz, "%*sCA:%s", indent, "",
+                                                        obj->ca ? isCa : notCa);
+                break;
+            }
+        case ALT_NAMES_OID:
+            {
+                WOLFSSL_STACK* sk;
+                char* val;
+                int len;
+                tmp[0] = '\0'; /* Make sure tmp is null-terminated */
+
+                sk = ext->ext_sk;
+                while (sk != NULL) {
+                    /* str is GENERAL_NAME for subject alternative name ext */
+                    str = sk->data.gn->d.ia5;
+                    len = str->length + 2; /* + 2 for NULL char and "," */
+                    val = (char*)XMALLOC(len + indent, NULL,
+                                                       DYNAMIC_TYPE_TMP_BUFFER);
+                    if (sk->next)
+                        XSNPRINTF(val, len, "%*s%s, ", indent, "", str->strData);
+                    else
+                        XSNPRINTF(val, len, "%*s%s", indent, "", str->strData);
+
+                    XSTRNCAT(tmp, val, len);
```

Comment:
I believe Jenkins is reporting a possible destination overflow case... make sure and check `len` does not exceed `tmp` size.